common/service/host/kcs.c

```diff
@@ -55,14 +60,90 @@ void reset_kcs_ok()
 	proc_kcs_ok = false;
 }
 
+#ifdef ENABLE_PLDM
+
+enum cmd_app_get_sys_info_params {
+	LENGTH_INDEX = 0x05, // skip netfun, cmd code, paramter selctor, set selctor, encoding
+	VERIONS_START_INDEX = 0x06,
+};
+
+static uint8_t init_text_string_value(const char *new_text_string, char *text_strings)
+{
+	const uint8_t NULL_BYTE = 1, EMPTY_TEXT_STRINGS_SIZE = 2;
+	const uint8_t original_size = pldm_smbios_get_text_strings_size(text_strings);
+	const uint8_t new_text_string_len = strlen(new_text_string);
+	uint8_t new_text_string_start_idx = original_size - 1;
+
+	if (original_size == EMPTY_TEXT_STRINGS_SIZE) {
+		new_text_string_start_idx = 0;
+	}
+
+	text_strings = realloc(text_strings, original_size + new_text_string_len + NULL_BYTE);
+	memcpy(text_strings + new_text_string_start_idx, new_text_string, new_text_string_len);
+
+	*(text_strings + new_text_string_start_idx + new_text_string_len) = '\0';
+	*(text_strings + new_text_string_start_idx + new_text_string_len + 1) = '\0';
+	return pldm_smbios_get_text_strings_count(text_strings);
+}
+
+static void init_bios_information(smbios_bios_information *bios_info, char *bios_version)
+{
+	const uint8_t NOT_IMPLEMENTED = 0;
+
+	bios_info->header.type = SMBIOS_BIOS_INFORMATION;
+	bios_info->text_strings = malloc(sizeof(char) * 2);
```

Comment:
How is this freed?

Reply:
This is freed at the end of update_bios_information(), line 129
```
SAFE_FREE(bios_info->text_strings);
```